lib/src/usage.dart

```diff
@@ -171,8 +176,52 @@ class Usage {
     numHelpLines = 0;
   }
 
+  /// Wrap a single line of text into lines no longer than length.
+  /// Try to split at whitespace, but if that's not good enough to keep it
+  /// under the length, then split in the middle of a word.
+  List<String> wrap(String text, int length) {
+    assert(length > 0, 'Wrap length must be larger than zero.');
+    text = text.trim();
+    if (text.length <= length) {
+      return [text];
+    }
+    var result = <String>[];
+    int currentLineStart = 0;
+    int lastWhitespace;
+    for (int i = 0; i < text.length; ++i) {
+      if (isWhitespace(text, i)) {
+        lastWhitespace = i;
+      }
+      if ((i - currentLineStart) >= length) {
+        // Back up to the last whitespace, unless there wasn't any, in which
+        // case we just split where we are.
+        if (lastWhitespace != null) {
+          i = lastWhitespace;
+        }
+        result.add(text.substring(currentLineStart, i));
+        // Skip any intervening whitespace.
+        while (isWhitespace(text, i) && i < text.length) i++;
+        currentLineStart = i;
+        lastWhitespace = null;
+      }
+    }
+    result.add(text.substring(currentLineStart));
+    return result;
+  }
+
   void write(int column, String text) {
+    const int minColumnWidth = 10;
     var lines = text.split('\n');
+    if (column == columnWidths.length && maxLineLength != null) {
+      var wrappedLines = <String>[];
+      var start = 0;
+      columnWidths.sublist(0, column).forEach((int width) => start += width);
```

Comment:
I think this is a more idiomatic using `reduce()`. Or just a plain for-in loop.

Reply:
OK.  Switched to reduce().